### PR TITLE
Reserve generated names as well as user supplied

### PR DESCRIPTION
**Pull request by hickeng**
#6561 ensured that user supplied names were reserved during creation
but did not do the same for generated names. 
Build https://ci.vcna.io/vmware/vic/14611 failed due to a generated
name collision.

Followup #6261 #6561
Fixes: #6697 
Towards: #334 (still requires id reservation)

### CODE REVIEW

File: lib/apiservers/engine/backends/container.go [hunk 1]
```diff
@@ -592,8 +592,18 @@ func (c *Container) ContainerCreate(config types.ContainerCreateConfig) (contain
...
-		return containertypes.ContainerCreateCreatedBody{}, derr.NewRequestConflictError(err)
+	if config.Name != "" {
+		err := cache.ContainerCache().ReserveName(container, config.Name)
+		if err != nil {
+			return containertypes.ContainerCreateCreatedBody{}, derr.NewRequestConflictError(err)
+		}
+	} else {
+		for {
+			config.Name = namesgenerator.GetRandomName(0)
+			if cache.ContainerCache().ReserveName(container, config.Name) == nil {
```
Comment (anchal-agrawal) [inline lib/apiservers/engine/backends/container.go]: We should add a test case in the `1-04-Docker-Create` suite for this. There's already a test at https://github.com/vmware/vic/blob/master/tests/test-cases/Group1-Docker-Commands/1-04-Docker-Create.robot#L194 - we can add this case there.

Comment (hickeng) [reply lib/apiservers/engine/backends/container.go]: I am unaware of a means to manipulate urandom to force a collision - I will look at adding a unit test and mocking `GetRandomName`.

Comment (anchal-agrawal) [reply lib/apiservers/engine/backends/container.go]: Oh right, my bad!

**Comment by hickeng** [comment]
@anchal-agrawal I've updated it for a limited retry and using the numeric suffix mechanism.

**Comment by sflxn** [comment]
lgtm

**Pull request merged**